#30087 [CUBRID] Provide more information on Query Plan TableView

rathana-pvs opened the pull request:
**Description:**
- provide additional information

1. index: index name if index scan is used
2. terms: predicate
3. total (r/p): Rows and Pages to be fetched
4. extra: extra information

rathana-pvs committed:
#30087 [CUBRID] Provide more information on Query Plan TableView

Add more columns (index, terms, total, extra) on query plan tableview

E1izabeth commented:
Thank you for contribution and sorry for long waiting. We missed your PR.
Will try to review it shortly and let you know.

rathana-pvs commented:
> Thank you for contribution and sorry for long waiting. We missed your PR. Will try to review it shortly and let you know.

Thank you for checking.

rathana-pvs committed:
Merge branch 'devel' into query_plan_extra_column

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java:
```diff
@@ -33,11 +33,14 @@ public class CubridPlanNode extends AbstractExecutionPlanNode
-    private Number row;
+    private String index;
+    private String term;
+    private Long cost;
+    private Long row;
```
Please make these `long` and remove associated `@NotNull` annotations. This annotation is not required for primitives.

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java:
```diff
@@ -66,23 +69,47 @@ public String getNodeType() {
     @Property(order = 2, viewable = true)
-    public Number getCost() {
-        return cost;
+    public String getIndex(){
```
Please fix in other methods as well. You also are missing spaces in control-flow statements (e.g. `if(` -> `if (`)
```suggestion
    public String getIndex() {
```

ShadelessFox commented on plugins/org.jkiss.dbeaver.ext.cubrid/src/org/jkiss/dbeaver/ext/cubrid/model/plan/CubridPlanNode.java:
```diff
@@ -168,18 +200,45 @@ private void parseObject(@NotNull List<String> segments) {
         }
     }
     
+    @Nullable
+    private String getTermExtra(boolean isTerm) {
+        String[] values = terms.get(term).split(" \\(sel");
+        if(isTerm)
+            return values[0].trim();
+        else
+            return "(sel" + values[1].trim();
+    }
+
+    @Nullable
+    private String getNameOrTotal(boolean isName) {
+        Pattern p;
+        if(isName) {
+            p = Pattern.compile("\\w+ \\w+");
+        } else {
+            p = Pattern.compile("\\w+\\/\\w+");
+        }
+        Matcher m = p.matcher(classNode.get(nodeName));
+        if(m.find()) {
+            return m.group(0);
+        }
+        return null;
+    }
```
To be honest, I'd cache these. These methods will be called every time the property is read.

rathana-pvs committed:
#30087 Update format code.

rathana-pvs committed:
#30087 Chage from Long to primitive long remove annotation.

rathana-pvs committed:
Merge branch 'devel' into query_plan_extra_column

E1izabeth commented:
We passed your PR to out QA team for testing

uslss committed:
Merge branch 'devel' into query_plan_extra_column

uslss commented:
The error appears on Explain Execution Plan 
![image](https://github.com/user-attachments/assets/d0ff1978-e881-4eb9-b16b-2558ac82655b)
```

eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Mon Jul 15 16:18:39 CEST 2024
Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null

java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTermExtra(CubridPlanNode.java:203)
	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTerms(CubridPlanNode.java:84)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.lambda$1(ObjectPropertyDescriptor.java:431)
	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.readValue(ObjectPropertyDescriptor.java:460)
	at org.jkiss.dbeaver.runtime.properties.PropertySourceAbstract.getPropertyValue(PropertySourceAbstract.java:224)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl.getCellValue(ObjectListControl.java:764)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl$PaintListener.handleEvent(ObjectListControl.java:1278)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1184)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1169)
	at org.eclipse.swt.widgets.Tree.CDDS_ITEMPOSTPAINT(Tree.java:871)
	at org.eclipse.swt.widgets.Tree.wmNotifyChild(Tree.java:7558)
	at org.eclipse.swt.widgets.Control.wmNotify(Control.java:5778)
	at org.eclipse.swt.widgets.Composite.wmNotify(Composite.java:1965)
	at org.eclipse.swt.widgets.Control.WM_NOTIFY(Control.java:5346)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4768)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
	at org.eclipse.swt.internal.win32.OS.CallWindowProc(Native Method)
	at org.eclipse.swt.widgets.Tree.callWindowProc(Tree.java:1574)
	at org.eclipse.swt.widgets.Widget.wmPaint(Widget.java:2289)
	at org.eclipse.swt.widgets.Control.WM_PAINT(Control.java:5351)
	at org.eclipse.swt.widgets.Composite.WM_PAINT(Composite.java:1491)
	at org.eclipse.swt.widgets.Tree.WM_PAINT(Tree.java:7167)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4769)
	at org.eclipse.swt.widgets.Tree.windowProc(Tree.java:6148)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
	at org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3671)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at org.eclipse.ui.internal.views.log.EventDetailsDialog.open(EventDetailsDialog.java:198)
	at org.eclipse.ui.internal.views.log.EventDetailsDialogAction.run(EventDetailsDialogAction.java:102)
	at org.eclipse.ui.internal.views.log.LogView.lambda$2(LogView.java:624)
	at org.eclipse.jface.viewers.StructuredViewer$1.run(StructuredViewer.java:779)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.StructuredViewer.fireDoubleClick(StructuredViewer.java:776)
	at org.eclipse.jface.viewers.AbstractTreeViewer.handleDoubleSelect(AbstractTreeViewer.java:1580)
	at org.eclipse.jface.viewers.StructuredViewer$4.widgetDefaultSelected(StructuredViewer.java:1205)
	at org.eclipse.jface.util.OpenStrategy.fireDefaultSelectionEvent(OpenStrategy.java:271)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:328)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
eclipse.buildId=unknown
java.version=17.0.6
java.vendor=Eclipse Adoptium
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
Command-line arguments:  -os win32 -ws win32 -arch x86_64

org.jkiss.dbeaver.model
Error
Mon Jul 15 16:18:39 CEST 2024
Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null

java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTermExtra(CubridPlanNode.java:203)
	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getExtra(CubridPlanNode.java:106)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.lambda$1(ObjectPropertyDescriptor.java:431)
	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.readValue(ObjectPropertyDescriptor.java:460)
	at org.jkiss.dbeaver.runtime.properties.PropertySourceAbstract.getPropertyValue(PropertySourceAbstract.java:224)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl.getCellValue(ObjectListControl.java:764)
	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl$PaintListener.handleEvent(ObjectListControl.java:1278)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1184)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1169)
	at org.eclipse.swt.widgets.Tree.CDDS_ITEMPOSTPAINT(Tree.java:871)
	at org.eclipse.swt.widgets.Tree.wmNotifyChild(Tree.java:7558)
	at org.eclipse.swt.widgets.Control.wmNotify(Control.java:5778)
	at org.eclipse.swt.widgets.Composite.wmNotify(Composite.java:1965)
	at org.eclipse.swt.widgets.Control.WM_NOTIFY(Control.java:5346)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4768)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
	at org.eclipse.swt.internal.win32.OS.CallWindowProc(Native Method)
	at org.eclipse.swt.widgets.Tree.callWindowProc(Tree.java:1574)
	at org.eclipse.swt.widgets.Widget.wmPaint(Widget.java:2289)
	at org.eclipse.swt.widgets.Control.WM_PAINT(Control.java:5351)
	at org.eclipse.swt.widgets.Composite.WM_PAINT(Composite.java:1491)
	at org.eclipse.swt.widgets.Tree.WM_PAINT(Tree.java:7167)
	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4769)
	at org.eclipse.swt.widgets.Tree.windowProc(Tree.java:6148)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
	at org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3671)
	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
	at org.eclipse.jface.window.Window.open(Window.java:799)
	at org.eclipse.ui.internal.views.log.EventDetailsDialog.open(EventDetailsDialog.java:198)
	at org.eclipse.ui.internal.views.log.EventDetailsDialogAction.run(EventDetailsDialogAction.java:102)
	at org.eclipse.ui.internal.views.log.LogView.lambda$2(LogView.java:624)
	at org.eclipse.jface.viewers.StructuredViewer$1.run(StructuredViewer.java:779)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.StructuredViewer.fireDoubleClick(StructuredViewer.java:776)
	at org.eclipse.jface.viewers.AbstractTreeViewer.handleDoubleSelect(AbstractTreeViewer.java:1580)
	at org.eclipse.jface.viewers.StructuredViewer$4.widgetDefaultSelected(StructuredViewer.java:1205)
	at org.eclipse.jface.util.OpenStrategy.fireDefaultSelectionEvent(OpenStrategy.java:271)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:328)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)


```

rathana-pvs committed:
#30087 fix null exception when split string.

rathana-pvs commented:
> The error appears on Explain Execution Plan ![image](https://private-user-images.githubusercontent.com/31996417/348738959-d0ff1978-e881-4eb9-b16b-2558ac82655b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE0NTc1OTMsIm5iZiI6MTcyMTQ1NzI5MywicGF0aCI6Ii8zMTk5NjQxNy8zNDg3Mzg5NTktZDBmZjE5NzgtZTg4MS00ZWI5LWIxNmItMjU1OGFjODI2NTViLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzIwVDA2MzQ1M1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA0NzE2ZTdhYWQxOWNiNmMzNDM5MzIxZTc2ZTFhYTE2ODY1MmNlYzk3YzQ0NTE1OGNjNjFhOTFlM2U2ZWZmMWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.dGGQLAeJnjiu4DQqffuXV81w0habXZ5rqpzlZbhBPQ4)
> 
> ```
> 
> eclipse.buildId=unknown
> java.version=17.0.6
> java.vendor=Eclipse Adoptium
> BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
> Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
> Command-line arguments:  -os win32 -ws win32 -arch x86_64
> 
> org.jkiss.dbeaver.model
> Error
> Mon Jul 15 16:18:39 CEST 2024
> Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
> 
> java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
> 	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTermExtra(CubridPlanNode.java:203)
> 	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTerms(CubridPlanNode.java:84)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
> 	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.lambda$1(ObjectPropertyDescriptor.java:431)
> 	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.readValue(ObjectPropertyDescriptor.java:460)
> 	at org.jkiss.dbeaver.runtime.properties.PropertySourceAbstract.getPropertyValue(PropertySourceAbstract.java:224)
> 	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl.getCellValue(ObjectListControl.java:764)
> 	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl$PaintListener.handleEvent(ObjectListControl.java:1278)
> 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
> 	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1184)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1169)
> 	at org.eclipse.swt.widgets.Tree.CDDS_ITEMPOSTPAINT(Tree.java:871)
> 	at org.eclipse.swt.widgets.Tree.wmNotifyChild(Tree.java:7558)
> 	at org.eclipse.swt.widgets.Control.wmNotify(Control.java:5778)
> 	at org.eclipse.swt.widgets.Composite.wmNotify(Composite.java:1965)
> 	at org.eclipse.swt.widgets.Control.WM_NOTIFY(Control.java:5346)
> 	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4768)
> 	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
> 	at org.eclipse.swt.internal.win32.OS.CallWindowProc(Native Method)
> 	at org.eclipse.swt.widgets.Tree.callWindowProc(Tree.java:1574)
> 	at org.eclipse.swt.widgets.Widget.wmPaint(Widget.java:2289)
> 	at org.eclipse.swt.widgets.Control.WM_PAINT(Control.java:5351)
> 	at org.eclipse.swt.widgets.Composite.WM_PAINT(Composite.java:1491)
> 	at org.eclipse.swt.widgets.Tree.WM_PAINT(Tree.java:7167)
> 	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4769)
> 	at org.eclipse.swt.widgets.Tree.windowProc(Tree.java:6148)
> 	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
> 	at org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
> 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3671)
> 	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
> 	at org.eclipse.jface.window.Window.open(Window.java:799)
> 	at org.eclipse.ui.internal.views.log.EventDetailsDialog.open(EventDetailsDialog.java:198)
> 	at org.eclipse.ui.internal.views.log.EventDetailsDialogAction.run(EventDetailsDialogAction.java:102)
> 	at org.eclipse.ui.internal.views.log.LogView.lambda$2(LogView.java:624)
> 	at org.eclipse.jface.viewers.StructuredViewer$1.run(StructuredViewer.java:779)
> 	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
> 	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
> 	at org.eclipse.jface.viewers.StructuredViewer.fireDoubleClick(StructuredViewer.java:776)
> 	at org.eclipse.jface.viewers.AbstractTreeViewer.handleDoubleSelect(AbstractTreeViewer.java:1580)
> 	at org.eclipse.jface.viewers.StructuredViewer$4.widgetDefaultSelected(StructuredViewer.java:1205)
> 	at org.eclipse.jface.util.OpenStrategy.fireDefaultSelectionEvent(OpenStrategy.java:271)
> 	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:328)
> 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
> 	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
> 	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
> 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
> 	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
> 	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
> 	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
> 	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
> 	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
> 	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
> 	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
> 	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
> 	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
> 	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
> 	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
> 	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
> 	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
> 	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
> eclipse.buildId=unknown
> java.version=17.0.6
> java.vendor=Eclipse Adoptium
> BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=en
> Framework arguments:  -eclipse.keyring C:\Users\Dziyana\AppData\Roaming\DBeaverData\secure\secure_storage
> Command-line arguments:  -os win32 -ws win32 -arch x86_64
> 
> org.jkiss.dbeaver.model
> Error
> Mon Jul 15 16:18:39 CEST 2024
> Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
> 
> java.lang.NullPointerException: Cannot invoke "String.split(String)" because the return value of "java.util.Map.get(Object)" is null
> 	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getTermExtra(CubridPlanNode.java:203)
> 	at org.jkiss.dbeaver.ext.cubrid.model.plan.CubridPlanNode.getExtra(CubridPlanNode.java:106)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
> 	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.lambda$1(ObjectPropertyDescriptor.java:431)
> 	at org.jkiss.dbeaver.runtime.properties.ObjectPropertyDescriptor.readValue(ObjectPropertyDescriptor.java:460)
> 	at org.jkiss.dbeaver.runtime.properties.PropertySourceAbstract.getPropertyValue(PropertySourceAbstract.java:224)
> 	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl.getCellValue(ObjectListControl.java:764)
> 	at org.jkiss.dbeaver.ui.navigator.itemlist.ObjectListControl$PaintListener.handleEvent(ObjectListControl.java:1278)
> 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
> 	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1184)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1169)
> 	at org.eclipse.swt.widgets.Tree.CDDS_ITEMPOSTPAINT(Tree.java:871)
> 	at org.eclipse.swt.widgets.Tree.wmNotifyChild(Tree.java:7558)
> 	at org.eclipse.swt.widgets.Control.wmNotify(Control.java:5778)
> 	at org.eclipse.swt.widgets.Composite.wmNotify(Composite.java:1965)
> 	at org.eclipse.swt.widgets.Control.WM_NOTIFY(Control.java:5346)
> 	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4768)
> 	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
> 	at org.eclipse.swt.internal.win32.OS.CallWindowProc(Native Method)
> 	at org.eclipse.swt.widgets.Tree.callWindowProc(Tree.java:1574)
> 	at org.eclipse.swt.widgets.Widget.wmPaint(Widget.java:2289)
> 	at org.eclipse.swt.widgets.Control.WM_PAINT(Control.java:5351)
> 	at org.eclipse.swt.widgets.Composite.WM_PAINT(Composite.java:1491)
> 	at org.eclipse.swt.widgets.Tree.WM_PAINT(Tree.java:7167)
> 	at org.eclipse.swt.widgets.Control.windowProc(Control.java:4769)
> 	at org.eclipse.swt.widgets.Tree.windowProc(Tree.java:6148)
> 	at org.eclipse.swt.widgets.Display.windowProc(Display.java:5050)
> 	at org.eclipse.swt.internal.win32.OS.DispatchMessage(Native Method)
> 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3671)
> 	at org.eclipse.jface.window.Window.runEventLoop(Window.java:823)
> 	at org.eclipse.jface.window.Window.open(Window.java:799)
> 	at org.eclipse.ui.internal.views.log.EventDetailsDialog.open(EventDetailsDialog.java:198)
> 	at org.eclipse.ui.internal.views.log.EventDetailsDialogAction.run(EventDetailsDialogAction.java:102)
> 	at org.eclipse.ui.internal.views.log.LogView.lambda$2(LogView.java:624)
> 	at org.eclipse.jface.viewers.StructuredViewer$1.run(StructuredViewer.java:779)
> 	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:47)
> 	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
> 	at org.eclipse.jface.viewers.StructuredViewer.fireDoubleClick(StructuredViewer.java:776)
> 	at org.eclipse.jface.viewers.AbstractTreeViewer.handleDoubleSelect(AbstractTreeViewer.java:1580)
> 	at org.eclipse.jface.viewers.StructuredViewer$4.widgetDefaultSelected(StructuredViewer.java:1205)
> 	at org.eclipse.jface.util.OpenStrategy.fireDefaultSelectionEvent(OpenStrategy.java:271)
> 	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:328)
> 	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:91)
> 	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:4285)
> 	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1160)
> 	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4083)
> 	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3673)
> 	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$5.run(PartRenderingEngine.java:1151)
> 	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
> 	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1042)
> 	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:152)
> 	at org.eclipse.ui.internal.Workbench.lambda$3(Workbench.java:639)
> 	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:339)
> 	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:546)
> 	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:173)
> 	at org.jkiss.dbeaver.ui.app.standalone.DBeaverApplication.start(DBeaverApplication.java:291)
> 	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:208)
> 	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:143)
> 	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:109)
> 	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:439)
> 	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:271)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
> 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
> 	at java.base/java.lang.reflect.Method.invoke(Unknown Source)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.invokeFramework(DBeaverLauncher.java:648)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.basicRun(DBeaverLauncher.java:608)
> 	at org.jkiss.dbeaver.launcher.DBeaverLauncher.run(DBeaverLauncher.java:1459)
> ```

Thank you for check it. 
Now, I fixed it.

uslss committed:
Merge branch 'devel' into query_plan_extra_column

uslss commented:
verified
